### PR TITLE
feat: feedback tab - improve the UI and switch to HogQL

### PR DESCRIPTION
**Pull request by lharries**
- Feedback page: improves the UI, switches to using the query for both the data table and the trends
- User interview scheduler: adds the ability to view the flags for scheduling and adds a modal for easily creating a new interview

TODO before merging:

- [x] Get the feedback form errors to show up in red, for some reason the styling seems off

https://user-images.githubusercontent.com/22766134/225634321-4216b295-ebbb-413e-9659-e74a9ba10597.mp4

Screenshot Luke Harries  Luke Harries's Blog (Google Chrome) 2023-03-16 at 13 39@2x

Screenshot Feedback • PostHog (Google Chrome) 2023-03-16 at 13 40@2x

Screenshot Feedback • PostHog (Google Chrome) 2023-03-16 at 13 40 3@2x

Screenshot Feedback • PostHog (Google Chrome) 2023-03-16 at 13 40 2@2x

### CODE REVIEW

**Comment by lharries** [comment]
@Twixes / @pauldambra I can't seem to work out why the form isn't styling properly? Any suggestions would be great otherwise I can dive into it later on

**Comment by Twixes** [comment]
So, you were using the plain back-office Kea `Field`, but our actual styled version is in `lib/forms/Field`. Switched to the latter :)

Screenshot 2023-03-16 at 16 49 04

**Comment by lharries** [comment]
Great thank you!

File: frontend/src/scenes/feedback/userInterviewSchedulerLogic.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        ],
+    }),
+    forms(({ values }) => ({
+        interviewFlag: {
+            defaults: {
+                key: 'interview-' as string,
+                title: 'Help improve our product!' as string,
+                body: 'We would love to hear your feedback on a short call' as string,
+                description: 'Description' as string,
+                bookingLink: 'https://calendly.com/...' as string,
```
Comment (Twixes) [inline frontend/src/scenes/feedback/userInterviewSchedulerLogic.ts:45]: This kinda looks like an ad for Calendly 😄

File: frontend/src/scenes/feedback/userInterviewSchedulerLogic.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            false as boolean,
+            {
+                toggleInterviewFlagModal: (state) => !state,
+            },
+        ],
+    }),
+    forms(({ values }) => ({
+        interviewFlag: {
+            defaults: {
+                key: 'interview-' as string,
```
Comment (Twixes) [inline frontend/src/scenes/feedback/userInterviewSchedulerLogic.ts:41]:
```suggestion
                key: FLAG_PREFIX as string,
```

File: frontend/src/scenes/feedback/inAppFeedbackLogic.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            [] as EventType[],
+            {
+                loadEvents: async ({ eventName }: { eventName: string }) => {
+                    const response = await api.events.list({
+                        properties: [],
+                        event: eventName,
+                        orderBy: ['-timestamp'],
+                    })
+                    // TODO: fix this type
+                    return response.results as unknown as EventType[]
```
Comment (Twixes) [inline frontend/src/scenes/feedback/inAppFeedbackLogic.ts:111]: This is fixed now on `master`, the `as` can be removed

File: frontend/src/scenes/feedback/Feedback.tsx [hunk 4]
```diff
@@ -271,13 +24,26 @@ export const Feedback = (): JSX.Element => {
...
+                        key: 'in-app-feedback',
+                        label: 'In-app feedback',
+                        tooltip: 'Analyze feedback from your users',
+                    },
+                    {
+                        content: <UserInterviewScheduler />,
+                        key: 'user-interview-scheduler',
+                        label: 'User interview scheduler',
+                        tooltip: 'Schedule user interviews with your users',
+                    },
```
Comment (Twixes) [inline frontend/src/scenes/feedback/Feedback.tsx:45]: These tooltips don't seem very useful. There's always a cost in terms of mental load when adding an info icon + tooltip, so we should only add them when they can really clarify something. It does not make things particularly clearer to say that an interview scheduler allows scheduling interview.

File: frontend/src/scenes/feedback/UserInterviewScheduler.tsx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+export function UserInterviewScheduler(): JSX.Element {
+    const { toggleInterviewFlagModal, toggleSchedulerInstructions } = useActions(userInterviewSchedulerLogic)
+
+    return (
+        <div>
+            <div className="flex w-full justify-between">
+                <div>
+                    <h3 className="text-lg">User Interview Scheduler</h3>
```
Comment (Twixes) [inline frontend/src/scenes/feedback/UserInterviewScheduler.tsx:187]: The active tab already serves as a sub-heading, so this looks redundant

File: frontend/src/scenes/feedback/UserInterviewScheduler.tsx [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+        </LemonModal>
+    )
+}
+
+export function CreateInterviewFlag(): JSX.Element {
+    const { interviewFlagModal, isInterviewFlagSubmitting } = useValues(userInterviewSchedulerLogic)
+
+    const { toggleInterviewFlagModal } = useActions(userInterviewSchedulerLogic)
+
+    // TODO: fix form styling, errors don't seem to be the right color and the overal style looks off
```
Comment (Twixes) [inline frontend/src/scenes/feedback/UserInterviewScheduler.tsx:151]:
```suggestion
```

**Review by Twixes — COMMENTED** [review]
This is definitely getting nicer! A few comments below

**Comment by lharries** [comment]
@Twixes good suggestions! I've implemented them all. Let me know if you think there's anything else blocking the merge

**Review by Twixes — APPROVED** [review]

**Pull request merged**